core/src/main/java/org/bitcoinj/core/VersionedChecksummedBytes.java

```diff
@@ -33,29 +36,20 @@
  * <p>and the result is then Base58 encoded. This format is used for addresses, and private keys exported using the
  * dumpprivkey command.</p>
  */
-public class VersionedChecksummedBytes implements Serializable, Cloneable, Comparable<VersionedChecksummedBytes> {
```

Comment:
Could we add an abstract `getFormat()` and an enum like this:

```
enum Format {
    BASE58,
    BECH32
}
```

When we handle the abstract types, it's handy to know what is the encoding format.

Reply:
In a way, I wanted to abstract away the encoding from the user. Just use toString() and you'll get the correct encoding.

But of course if it's important, we can add it. You could also inspect the type though.

Reply:
OK, in that case we can leave it.